list-agreements.lua

```diff
@@ -1,120 +1,218 @@
+    local agr_year_tick = agr.details[0].year_tick
+    local julian_day = math.floor(agr_year_tick / 1200) + 1
+    local agr_month = math.floor(julian_day / 28) + 1
+    local agr_day = julian_day % 28
+    return string.format("%03d-%02d-%02d",agr_year, agr_month, agr_day)
 end
 
-function get_petition_date(agr)
-    agr_year = agr.details[0].year
-    agr_year_tick = agr.details[0].year_tick
-    julian_day = math.floor(agr_year_tick / 1200) + 1
-    agr_month = math.floor(julian_day / 28) + 1
-    agr_day = julian_day % 28
-    return agr_year.."-"..agr_month.."-"..agr_day
+function get_petition_age(agr)
+    local agr_year_tick = agr.details[0].year_tick
+    local agr_year = agr.details[0].year
+    local cur_year_tick = df.global.cur_year_tick
+    local cur_year = df.global.cur_year
+    local del_year, del_year_tick, del_month, del_day
+    --delta, check to prevent off by 1 error, not validated
+    if cur_year_tick > agr_year_tick then
+        del_year = cur_year - agr_year
+        del_year_tick = cur_year_tick - agr_year_tick
+    else
+        del_year = cur_year - agr_year - 1
+        del_year_tick = agr_year_tick - cur_year_tick
+    end
+    local julian_day = math.floor(del_year_tick / 1200) + 1
+    del_month = math.floor(julian_day / 28)
+    del_day = julian_day % 28
+    return {del_year,del_month,del_day}
 end
 
 function get_guildhall_profession(agr)
-    prof = agr.details[0].data.Location.profession
-    profname = string.lower(df.profession[prof])
+    local prof = agr.details[0].data.Location.profession
+    local profname = string.lower(df.profession[prof])
+    -- *VERY* important code follows
+    if string.find(profname, "man") then
+        profname = string.gsub(profname,"man",string.lower(dfhack.units.getRaceNameById(df.global.ui.race_id)))
+    end
     return profname:gsub("_", " ")
 end
 
-function get_religion_name(agr)
-    religion_id = agr.details[0].data.Location.deity_data.Religion
-    religion_name = dfhack.TranslateName(df.global.world.entities.all[religion_id].name, true)
-    return religion_name
+function get_agr_party_name(agr)
+    --assume party 0 is guild/order, 1 is local government as siteid = playerfortid
+    local party_id = agr.parties[0].entity_ids[0]
+    local party_name = dfhack.TranslateName(df.global.world.entities.all[party_id].name, true)
+    return party_name
 end
 
-function is_satisfied(agr)
-    satisfied = agr.flags.convicted_accepted
-    return satisfied
+function get_deity_name(agr)
+    local religion_id = agr.details[0].data.Location.deity_data.Religion
+    for _,deity_id in ipairs(df.global.world.entities.all[religion_id].relations.deities) do
+        return dfhack.TranslateName(df.global.world.history.figures[deity_id].name,true)
+    end
 end
 
-function is_denied(agr)
-    denied = agr.flags.petition_not_accepted
-    return denied
+
+--return resolution status
+-- 0 no resolution
+-- 1 satisfied
+-- 2 denied
+-- 3 expired
```

Comment:
instead of returning an arbitrary numeric code, how about returning a boolean and a descriptive string that can be used directly? That way you can call it as
`if is_resolved(arg) then`
or
`local resolved, resolved_str = is_resolved(arg)`
which might simplify the calling code below